server/api/account.js

```diff
@@ -1,75 +1,99 @@
+        });
+    });
+}
+
+module.exports.init = function(server) {
+    server.post('/api/account/register', function(req, res, next) {
+        userRepository.getUserByUsername(req.body.username).then(user => {
+            if(user) {
                 res.send({ success: false, message: 'An account with that name already exists, please choose another' });
                 return next();
             }
 
-            bcrypt.hash(req.body.password, 10, function(err, hash) {
-                if(err) {
-                    res.send({ success: false, message: 'An error occured registering your account' });
-                    logger.info(err.message);
-                    return next(err);
-                }
-
-                req.body.password = hash;
-                req.body.registered = new Date();
-                req.body.emailHash = crypto.createHash('md5').update(req.body.email).digest('hex');
-
-                db.collection('users').insert(req.body, function(err) {
-                    if(err) {
-                        res.send({ success: false, message: 'An error occured registering your account' });
-                        logger.info(err.message);
-                        return next(err);
-                    }
-
-                    req.login(req.body, function(err) {
-                        if(err) {
-                            res.send({ success: false, message: 'An error occured registering your account' });
-                            logger.info(err.message);
-                            return next(err);
-                        }
-
-                        res.send({ success: true, user: req.body, token: jwt.sign(req.user, config.secret)});
-                    });
-                });
-            });
+            return hashPassword(req.body.password, 10);
+        }).then(hash => {
+            req.body.password = hash;
+            req.body.registered = new Date();
+            req.body.emailHash = crypto.createHash('md5').update(req.body.email).digest('hex');
+
+            return userRepository.addUser(req.body);
+        }).then(() => {
+            return loginUser(req, req.body);
+        }).then(() => {
+            res.send({ success: true, user: req.body, token: jwt.sign(req.user, config.secret)});
+        }).catch(err => {
+            res.send({ success: false, message: 'An error occured registering your account' });
+            logger.info(err.message);
+            return next(err);
         });
     });
 
     server.post('/api/account/check-username', function(req, res) {
-        db.collection('users').findOne({ username: { '$regex': new RegExp('^' + escapeRegex(req.body.username.toLowerCase()), 'i')}},
-        function(err, account) {
-            if(err) {
-                res.send({ message: '' });
-                logger.info(err.message);
-                return;
-            }
+        userRepository.getUserByUsername(req.body.username).then(user => {
+            logger.info(user);
```

Comment:
Did you intend to keep this log message here? If so, printing out context information in addition to the user object would be useful.